packages/engine/Specs/Scene/Model/RuntimeModelInstancingPipelineStageSpec.js

```diff
@@ -0,0 +1,289 @@
+import {
+  Cartesian3,
+  combine,
+  GltfLoader,
+  RuntimeModelInstancingPipelineStage,
+  ModelInstancesUpdateStage,
+  Math as CesiumMath,
+  Buffer,
+  BufferUsage,
+  ModelStatistics,
+  Resource,
+  ResourceCache,
+  SceneMode,
+  ShaderBuilder,
+  ModelInstance,
+  Transforms,
+  HeadingPitchRoll,
+  Ellipsoid,
+} from "../../../index.js";
+import createScene from "../../../../../Specs/createScene.js";
+import waitForLoaderProcess from "../../../../../Specs/waitForLoaderProcess.js";
+import ShaderBuilderTester from "../../../../../Specs/ShaderBuilderTester.js";
+
+describe(
+  "Scene/Model/RuntimeModelInstancingPipelineStage",
+  function () {
+    const sampleGltfUrl =
+      "./Data/Models/glTF-2.0/CesiumMilkTruck/glTF/CesiumMilkTruck.glb";
+
+    let scene;
+    let scene2D;
+    const gltfLoaders = [];
+
+    beforeAll(function () {
+      scene = createScene();
+      scene.renderForSpecs();
+
+      scene2D = createScene();
+      scene2D.renderForSpecs();
+    });
+
+    afterAll(function () {
+      scene = createScene();
+      scene2D.destroyForSpecs();
+    });
+
+    afterEach(function () {
+      const gltfLoadersLength = gltfLoaders.length;
+      for (let i = 0; i < gltfLoadersLength; ++i) {
+        const gltfLoader = gltfLoaders[i];
+        if (!gltfLoader.isDestroyed()) {
+          gltfLoader.destroy();
+        }
+      }
+      gltfLoaders.length = 0;
+      ResourceCache.clearForSpecs();
+    });
+
+    const samplePosition1 = new Cartesian3(10, 10, 10);
+    const samplePosition2 = new Cartesian3(20, 20, 20);
+
+    const headingPositionRoll = new HeadingPitchRoll();
+    const fixedFrameTransform = Transforms.localFrameToFixedFrameGenerator(
+      "north",
+      "west",
+    );
+    const instanceModelMatrix1 = new Transforms.headingPitchRollToFixedFrame(
+      samplePosition1,
+      headingPositionRoll,
+      Ellipsoid.WGS84,
+      fixedFrameTransform,
+    );
+
+    const instanceModelMatrix2 = new Transforms.headingPitchRollToFixedFrame(
+      samplePosition2,
+      headingPositionRoll,
+      Ellipsoid.WGS84,
+      fixedFrameTransform,
+    );
+
+    const sampleInstance1 = new ModelInstance(instanceModelMatrix1);
+    const sampleInstance2 = new ModelInstance(instanceModelMatrix2);
+
+    function mockRenderResources(node) {
+      return {
+        attributeIndex: 1,
+        attributes: [],
+        instancingTranslationMax: undefined,
+        instancingTranslationMin: undefined,
+        shaderBuilder: new ShaderBuilder(),
+        model: {
+          _modelResources: [],
+          _pipelineResources: [],
+          statistics: new ModelStatistics(),
+          sceneGraph: {
```

Comment:
Let's have this function only set up the mocks unrelated to the functionality we're testing. For instance, would probably expected to see something like the following line in the actual test block. 

````js
renderResources.sceneGraph.modelInstances = [sampleInstance1, sampleInstance2];
```` 

1. It helps make the test more readable for those who are working with them later. 
2. It allows the mock and the test code to change independently.